cpp/arcticdb/version/test/test_version_map_batch.cpp

```diff
@@ -0,0 +1,84 @@
+#include <gtest/gtest.h>
+
+#include <arcticdb/version/version_map_batch_methods.hpp>
+#include <arcticdb/stream/test/stream_test_common.hpp>
+
+struct VersionMapBatchStore : arcticdb::TestStore {
+protected:
+    std::string get_name() override {
+        return "version_map_batch";
+    }
+};
+
+namespace arcticdb {
+
+AtomKey test_index_key(StreamId id, VersionId version_id) {
+    return atom_key_builder().version_id(version_id).creation_ts(PilotedClock::nanos_since_epoch()).content_hash(3)
+        .start_index(4).end_index(5).build(id, KeyType::TABLE_INDEX);
+}
+
+void add_versions_for_stream(
+    std::shared_ptr<VersionMap> version_map,
+    std::shared_ptr<Store> store,
+    StreamId stream_id,
+    size_t num_versions,
+    size_t start = 0u) {
+    for(auto i = start; i < start + num_versions; ++i) {
+        version_map->write_version(store, test_index_key(stream_id, i));
+    }
+};
+}
+
+TEST_F(VersionMapBatchStore, Simple) {
+    using namespace arcticdb;
+    using namespace arcticdb::pipelines;
+
+    auto store = test_store_->_test_get_store();
+    auto version_map = std::make_shared<VersionMap>();
+
+    for(auto i = 0u; i < 5; ++i) {
+        auto stream = fmt::format("stream_{}", i);
+        add_versions_for_stream(version_map, store, stream, 5);
+    }
+
```

Comment:
was the original intention to differentiate between actions across these two loops?

Reply:
fixed in another commit

Reply:
simplified in another commit